### ISSUE TITLE
I can't try-catch deleteMessage with messageId "message->replyToMessage->messageId"

### ISSUE THREAD

**Issue by Amirhan-Taipovjan-Greatest-I** [reo7sp/tgbot-cpp#239]
>     try {
>         bot.getEvents().onCommand("delete", [&bot](Message::Ptr message) {
>             if (StringTools::startsWith(message->text, "/start")) {
>                 return;
>             }
>             int numb = rand50();
>             if (numb == 1) {
>                 try {
>                     bot.getApi().deleteMessage(message->chat->id, message->replyToMessage->messageId);
>                 }
>                 catch (exception& e) {
>                     printf("XATA BULMADY: %s\n", e.what());
>                 }
>             }
>             else bot.getApi().sendAnimation(message->chat->id, InputFile::fromFile("examples/cyrt.gif", "image/gif"), 0, 0, 0, "", "", message->messageId);
>             });
>     }
>     catch (exception& e) {
>         printf("XATA BULMADY: %s\n", e.what());
>     }

Even with try-catch block I keep getting this exception message like this:
![image](https://user-images.githubusercontent.com/51203385/198349494-6c134d77-c9bd-43dc-99d3-139b8c33ce0b.png)

**Comment by llnulldisk** [comment]
Yes, there is an error because replyToMessage (the already deleted message in this case) doesn't exist and points to zero. You would first have to check whether the message that is being replied to (replyToMessage) is valid at all and if so, then take the id of it.

**Comment by reo7sp** [comment]
`message->replyToMessage` can be null, and dereferencing it causes the access violation before `deleteMessage()` is called. Check it first: `if (message->replyToMessage) { ... }`.
